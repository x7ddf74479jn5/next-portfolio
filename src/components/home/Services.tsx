import Spacer from "src/components/layouts/Spacer";
import { services } from "src/contents/services";

import { NextArrowButton } from "../common/NextArrowButton";

const Services = () => {
  const slideInLeft = {
    class: "p-grid__list-item-floated animated slow slideInLeft",
    dataAnimate: "slideInLeft",
  };
  const slideInRight = {
    class: "p-grid__list-item-floated animated slow slideInRight",
    dataAnimate: "slideInRight",
  };
  return (
    <section className="c-section c-bg-primary">
      <div className="c-section-wrapin">
        <h2 className="p-headline__top">Services</h2>
        <hr className="c-section__line" />
        <Spacer size="sm" />
        <NextArrowButton label="事業内容" href="/about/#skills" />
        <Spacer size="md" />
        <div className="p-grid__list">
          {services.map((service, index) => {
            return (
              <article
                key={index}
                className={index % 2 === 0 ? slideInLeft.class : slideInRight.class}
                data-animate={index % 2 === 0 ? slideInLeft.dataAnimate : slideInRight.dataAnimate}
              >
                <h3>{service.title}</h3>
                <small>{service.caption}</small>
                <img className="p-media__thumb" src={service.img.src} alt={service.img.alt} />
                <p>{service.description}</p>
              </article>
            );
          })}
        </div>
      </div>
    </section>
  );
};

export default Services;
