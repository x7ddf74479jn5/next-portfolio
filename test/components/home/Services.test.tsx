import Services from "src/components/home/Services";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Services", () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<Services />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.getByRole("heading", { level: 2 })).toHaveTextContent("Services");
    expect(renderResult.getByRole("separator")).toBeInTheDocument();
    expect(renderResult.getByRole("button")).toHaveTextContent("事業内容");
    expect(renderResult.getByRole("button")).toHaveAttribute("href", "/about#skills");
  });
});
