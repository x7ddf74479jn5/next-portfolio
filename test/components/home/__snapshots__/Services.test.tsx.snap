// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section bgPrimary"
  >
    <div
      class="sectionWrapIn"
    >
      <h2
        class="headlineTop"
      >
        Services
      </h2>
      <hr
        class="sectionLine"
      />
      <div
        class="sm"
      />
      <a
        class="btnRound iconArrowNext"
        href="/about#skills"
        role="button"
      >
        事業内容
      </a>
      <div
        class="md"
      />
      <div
        class="gridList"
      >
        <article
          class="gridListItemFloated"
        >
          <h3>
            WEBサイト制作
          </h3>
          <small>
            Website Production
          </small>
          <img
            alt="Website Production"
            class="mediaThumb"
            height="288"
            src="/img/services/website-production.jpeg"
            width="512"
          />
          <p>
            コーポレートサイト、個人ポートフォリオ、ランディングページなどWebサイト制作全般を承ります。検索流入やSNS流入を強化する「SEOを意識したサイト」を作成できます。
          </p>
        </article>
        <article
          class="gridListItemFloated"
        >
          <h3>
            WordPress
          </h3>
          <small>
            WordPress
          </small>
          <img
            alt="WordPress"
            class="mediaThumb"
            height="288"
            src="/img/services/wordpress-icon.png"
            width="512"
          />
          <p>
            WordPress導入サポートやカスタマイズ、その他お困り事のご相談にも対応いたします。
          </p>
        </article>
        <article
          class="gridListItemFloated"
        >
          <h3>
            チャットアプリ開発
          </h3>
          <small>
            Chat APP
          </small>
          <img
            alt="Chat APP"
            class="mediaThumb"
            height="288"
            src="/img/services/web-app.jpeg"
            width="512"
          />
          <p>
            リアルタイム性が重視されるSNSアプリやチャットアプリなどを開発できます。これまで開発してきたWebアプリの一部を制作事例として公開しています。
          </p>
        </article>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
